src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -229,7 +229,7 @@ private ConstantValue DefaultSyntaxValue
                 // that never expires, then the failure mode will be to spin wait forever.
                 // For debug purposes let's instead use a token which expires after a modest amount of time
                 // to wait for the default syntax value to be available.
-                var cts = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+                var cts = new CancellationTokenSource(TimeSpan.FromMinutes(1));
```

Comment:
Was looking at this and thinking that even one minute probably isn't enough here. Seen CI have hiccups where threads stalled for considerable amounts of time in the past. 

This lead me to think about how this problem is approached elsewhere in our code base. Specifically how is `SpinWaitComplete` handled when called outside a `ForceComplete` method chain where there is an explicit `CancellationToken` to use. Found the following examples:

- `AliasSymbol.GetTarget`
- `SourceMemberContainerSymbol.GetMembersByNameSlow`

In these cases we always pass `default` for the `CancellationToken`. I think that is the approach we should take here. This means that well written code will behave correctly, even under extreme CPU load. While the current approach could exhibit incorrect behavior under extreme load. 

The upside is that when this is done wrong, if we make the mistake you mention, the code will deadlock and deadlocks are very easy to debug. :smile:

Reply:
Have simplified it to resemble the two examples you pointed out. I found it useful to introduce for debugging, but not worthwhile to attempt to justify the soundness of it (even if only in debug builds). Thanks!